File: server/Models/Note.cs
using System.ComponentModel.DataAnnotations;

namespace allspice.Models;

public class Note : RepoItem<int>
{
  public int RecipeId { get; set; }
  [MaxLength(5000)] public string Body { get; set; }
  [MaxLength(255)] public string CreatorId { get; set; }
}

File: client/src/models/Note.js
import { DatabaseItem } from "./DatabaseItem.js";


export class Note extends DatabaseItem{
  /**
   * @typedef NoteData
   * @property {number} id
   * @property {Date} createdAt
   * @property {Date} updatedAt
   * @property {number} recipeId
   * @property {string} body
   * @property {string} creatorId
   * @param {NoteData} data 
   */
  constructor(data){
    super(data)
    this.recipeId = data.recipeId
    this.body = data.body
    this.creatorId = data.creatorId
  }
}